### ISSUE TITLE
Softmax plus one for Flash attention

### ISSUE THREAD

**Issue by ethansmith2000** [pytorch/pytorch#120589]
### 🚀 The feature, motivation and pitch

After noticing some interesting behavior with attention in a little article i wrote [here](https://sweet-hall-e72.notion.site/Exploring-Cross-Attention-maps-in-Diffusion-Models-98a85f552fbe4e62887b896b756177f4?pvs=4): 

I wanted to see if using the method proposed [here](https://www.evanmiller.org/attention-is-off-by-one.html)
could suppress the behavior. However, memory-wise it isn't too feasible to train without flash attention, which is why I was wondering if it possible to implement their method for the kernel used in SDPA

### Alternatives

Non-flash attention although this is not feasible given token memory constraints

### Additional context

If it doesn't fit into torch, perhaps if you have any guidance on how I could make this change myself?

**Comment by mmmans** [comment]
I've implemented a flash-attn version here.  https://github.com/mmmans/flash-attention/commit/5928190d629b0d8664e0ae3b3b59f9b2a36d36c0
sample code is below:
output = flash_attn_func(query,key, value, softmax_offN=10)

**Comment by veritas9872** [comment]
I think that this would be a nice thing to have.
Although empirical verification has found that the +1 trick does not actually reduce outliers, it bothers me that we are using softmax in attention incorrectly.
